### PR TITLE
chore(deps): update dependency @types/node to v16

### PR DESCRIPTION
**Pull request by blumamir**
This is the same as #718 with an attempt to fix the failing CI jobs

Changes:
- Added more types on aws-sdk's `bindPromise` function. This function will be removed once the new messaging systems semantic conventions are released (along with other refactors to the SQS instrumentation)

### CODE REVIEW

**Comment by blumamir** [comment]
converted to draft until I'll be able to fix the typescript issues successfully.

**Comment by blumamir** [comment]
CI is now green, will appreciate another look at the typescript I added to make it pass.

File: plugins/node/opentelemetry-instrumentation-aws-sdk/src/utils.ts [hunk 1]
```diff
@@ -72,16 +72,24 @@ export const extractAttributesFromNormalizedRequest = (
...
-  target: Promise<any>,
+export const bindPromise = <T = unknown>(
+  target: Promise<T>,
   contextForCallbacks: Context,
   rebindCount = 1
-): Promise<any> => {
+): Promise<T> => {
   const origThen = target.then;
-  target.then = function (onFulfilled, onRejected) {
+  type PromiseThenParameters = Parameters<Promise<T>['then']>;
```
Comment (dyladan) [inline plugins/node/opentelemetry-instrumentation-aws-sdk/src/utils.ts:81]: 😮

**Review by dyladan — APPROVED** [review]
LGTM

**Review by Flarna — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-aws-sdk/src/utils.ts (hunk 1 above)
Comment (blumamir) [reply plugins/node/opentelemetry-instrumentation-aws-sdk/src/utils.ts:81]: is it good or bad? :)

Comment (blumamir) [reply plugins/node/opentelemetry-instrumentation-aws-sdk/src/utils.ts:81]: never had the chance to learn this advanced typescript stuff 😅

**Pull request merged**